### PR TITLE
Enable policy and actions signature verification with more unit tests

### PR DESCRIPTION
**Pull request by aleksmaus**
## What does this PR do?

Enabled the Agent policy and actions signature verification back for 8.9.0 releases.
Added more unit tests for the policy change and the app action handlers that were missing in the previous iteration.
This relates to the previous PR https://github.com/elastic/elastic-agent/pull/2348, the functionality was disabled for 8.8.0 release in order to reduce the surface.

## Why is it important?

This a part of Agent tamper protection.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)

## Related issues

- Relates to the previous PR https://github.com/elastic/elastic-agent/pull/2348

### CODE REVIEW

**Review by blakerouse — APPROVED** [review]
Looks really good and the testing is great to see, thanks for adding that.

**Comment by aleksmaus** [comment]
Was asked to hold off on merging this for now until further instructions from product.

**Comment by cmacknz** [comment]
Going to close this, reopen once we are ready to merge.

**Pull request closed without merging**
